src/Euclidean.jl

```diff
@@ -37,12 +45,45 @@ det_local_metric(M::MetricManifold{<:Manifold,EuclideanMetric}, x) = one(eltype(
 
 log_local_metric_density(M::MetricManifold{<:Manifold,EuclideanMetric}, x) = zero(eltype(x))
 
-inner(::Euclidean, x, v, w) = dot(v, w)
-inner(::MetricManifold{<:Manifold,EuclideanMetric}, x, v, w) = dot(v, w)
+@inline inner(::Euclidean, x, v, w) = dot(v, w)
+@inline inner(::MetricManifold{<:Manifold,EuclideanMetric}, x, v, w) = dot(v, w)
 
+distance(::Euclidean, x, y) = norm(x-y)
 norm(::Euclidean, x, v) = norm(v)
 norm(::MetricManifold{<:Manifold,EuclideanMetric}, x, v) = norm(v)
 
 exp!(M::Euclidean, y, x, v) = (y .= x + v)
 
 log!(M::Euclidean, v, x, y) = (v .= y - x)
+
+proj!(M::Euclidean, x) = x
```

Comment:
Well, I think it wasn't exported because we weren't sure it's the right name for it. `project_tangent` does a similar thing and has a longer name, so perhaps `project_point` would be a better name?

Reply:
That's a good point. I like `project_point!` for consistency as well.